### PR TITLE
Plotting

### PR DESCRIPTION
**Pull request by heikoklein**
New contour-plotting for ash-ensembles.
Fixing ISO3 codes outside map for nuclear-dispersion plots.

### CODE REVIEW

File: utils/SnapPy/snapEnsAshPlot.py [hunk 1]
```diff
@@ -2,27 +2,45 @@
...
                         message="Warning: 'partition' will ignore the 'mask' of the MaskedArray.")
 # shapefile.py uses root logger :-( and warns a lot about GSHHS
 import logging
+
 logging.root.setLevel(logging.ERROR)
 
-def plotMap(data, x, y, ax, title="", title_loc="center", clevs=[10,100,300,1000,3000,10000,30000,100000, 300000, 10000000], colors=None, extend='max'):
+
+def plotBasicMap(ax,title="", title_loc="center"):
+    """Plot multiple data on a map as contours with legends
```
Comment (charlienegri) [inline utils/SnapPy/snapEnsAshPlot.py:30]: why this docstring here? most arguments belong to PlotMap and PlotMapGrid

**Review by magnusuMET — APPROVED** [review]

File: utils/SnapPy/snapEnsAshPlot.py [hunk 2]
```diff
@@ -157,80 +173,94 @@ def snapens(ncfiles, hour, outfile):
...
-    #     )
-    # cbar = fig.colorbar(cs, cax=axins, format=formatter, orientation='vertical')
-    # cbar.set_label('mg/m³')
-
-    for i, (fli, flv) in enumerate(fl.items()):
-        ax = fig.add_subplot(4,3, i*3+1, projection=proj)
+    if contours_only == False:
+        fig = plt.figure(figsize=(12, 10.7))
+        fig.suptitle(f'{title}+{hour}hours ({endDT:%Y-%m-%d %H}:00Z). Uncertainty based upon {fl["350-550"].shape[0]} members.',
+            y=0.92)
```
[the remark this answers is not in the document]
Comment (heikoklein) [reply utils/SnapPy/snapEnsAshPlot.py:186]: y-0.92 is suptitle vertical position

File: utils/SnapPy/snapEnsAshPlot.py [hunk 3]
```diff
@@ -2,27 +2,45 @@
...
-from matplotlib import pyplot as plt
-import netCDF4
 import os
 import sys
 
+import netCDF4
 import numpy as np
+from matplotlib import pyplot as plt
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+from scipy.ndimage import gaussian_filter
```
[the remark this answers is not in the document]
Comment (heikoklein) [reply utils/SnapPy/snapEnsAshPlot.py:18]: scipy  is already part of the current conda environment in install_usermodule_r8.sh - though I don't know why (I guess through cartopy)? No change was needed.

File: utils/SnapPy/snapRunnerNpp [hunk 4]
```diff
@@ -87,10 +87,14 @@ def does_overlap(bbox, bboxlist):
...
     return False
 
-def get_country_plotlist(fig, ax, country_listname):
+def get_country_plotlist(fig, ax, country_listname, extent=None):
     countries = []
     textboxes = []
-    for country in get_country_list('europe2'):
+    for country in get_country_list(country_listname):
+        if extent:
+            if (country.lon < extent[0] or country.lon > extent[1] or
```
[the remark this answers is not in the document]
Comment (heikoklein) [reply utils/SnapPy/snapRunnerNpp:95]: <= needed to be exactly the same. But very elegant and good readable solution, which I didn't know  :-)

**Review by charlienegri — APPROVED** [review]

**Pull request merged**